augur/refine.py

```diff
@@ -211,15 +211,20 @@ def run(args):
             if n.name in metadata.index and 'date' in metadata.columns:
                 n.raw_date = metadata.loc[n.name, 'date']
 
-        tt = refine(tree=T, aln=aln, ref=ref, dates=dates, confidence=args.date_confidence,
-                    reroot=args.root, # or 'best', # We now have a default in param spec - this just adds confusion.
-                    Tc=0.01 if args.coalescent is None else args.coalescent, #use 0.01 as default coalescent time scale
-                    use_marginal = args.date_inference == 'marginal',
-                    branch_length_inference = args.branch_length_inference or 'auto',
-                    precision = 'auto' if args.precision is None else args.precision,
-                    clock_rate=args.clock_rate, clock_std=args.clock_std_dev,
-                    clock_filter_iqd=args.clock_filter_iqd,
-                    covariance=args.covariance, resolve_polytomies=(not args.keep_polytomies))
+        try:
+            tt = refine(tree=T, aln=aln, ref=ref, dates=dates, confidence=args.date_confidence,
+                        reroot=args.root, # or 'best', # We now have a default in param spec - this just adds confusion.
+                        Tc=0.01 if args.coalescent is None else args.coalescent, #use 0.01 as default coalescent time scale
+                        use_marginal = args.date_inference == 'marginal',
+                        branch_length_inference = args.branch_length_inference or 'auto',
+                        precision = 'auto' if args.precision is None else args.precision,
+                        clock_rate=args.clock_rate, clock_std=args.clock_std_dev,
+                        clock_filter_iqd=args.clock_filter_iqd,
+                        covariance=args.covariance, resolve_polytomies=(not args.keep_polytomies))
+        except BaseException as err:
+            print("ERROR: Was unable to refine time trees:\n", file=sys.stderr)
+            print(err, file=sys.stderr)
+            return 1
```

Comment:
Welcome @anna-parker! 🤗 

It totally makes sense for Augur to do a better job of handling treetime errors! It's not obvious in the refine command, but we've had a recent push to unify error handling in Augur since @victorlin added the [top-level catch of `AugurError`](https://github.com/nextstrain/augur/commit/2342f07b75f2ba9953cab8254ebef3d5febb5b5d).

Using `AugurError`, this can be simplified as:
```suggestion
            raise AugurError(f"Was unable to refine time trees:\n\n{err}")
```